mqtt/mqtt-bridge/src/persist/waking_state/ring_buffer/block.rs

```diff
@@ -141,24 +139,29 @@ impl BlockHeaderWithHash {
     }
 }
 
-/// A utility fn to calculate any entity that derives Hash.
-pub(crate) fn calculate_hash<T>(t: &T) -> u64
+/// A utility fn to calculate any crc over serialized objects.
+pub(crate) fn calculate_crc<T>(t: &T) -> BincodeResult<u32>
 where
-    T: Hash + ?Sized,
+    T: Serialize + ?Sized,
 {
-    let mut hasher = rustc_hash::FxHasher::default();
-    t.hash(&mut hasher);
-    hasher.finish()
+    let data = bincode::serialize(t)?;
```

Comment:
just an idea for future optimization: serialization allocates extra buffer. If we already have struct with stable and predictable layout (repr(C)), we could do some unsafe magic to transmute &T to &[u8]. But I believe we have to know the size of the `BlockVersion`. So, instead of calculating the size of `BlockHeaderWithCrc`, we can calculate the size of `BlockVersion` and add sizeof(u32) to get `BlockHeaderWithCrc` size.

I know @dmolokanov doesn't like unsafe, so this is just an option for the future, no need to fix it now. We have other low handing apples :)

Reply:
I think it is actually a good idea to avoid unnecessarily work

Reply:
Yeah, I have been trying to avoid unsafe magic like that if I can, but maybe in order to get good perf I will have to get my spellbooks out